docker/config/config.go

```diff
@@ -38,24 +38,44 @@ func (c *Config) IsEmpty() bool {
 // GetCredentials gets per-registry credentials from loaded Docker config
 func (c *Config) GetCredentials(registry string) (string, string, bool) {
 	if _, defined := c.usernames[registry]; !defined {
-		username, password, _ := credhelper.GetCredentials(registry, c.CredsStore, c.CredHelpers)
+		username, password, err := credhelper.GetCredentials(
+			registry,
+			c.CredsStore,
+			c.CredHelpers,
+		)
 
-		return username, password, false
+		if err != nil {
+			return "", "", false
+		}
+
+		return username, password, true
 	}
 
 	return c.usernames[registry], c.passwords[registry], true
 }
 
+func getAuthJSONString(username, password string) string {
+	if username == "_json_key" {
+		return fmt.Sprintf("%s:%s", username, password)
```

Comment:
Weird but OK as official Docker CLI does the same ..

Reply:
Weird indeed but this is how the things work.